inst/include/cpp11/integers.hpp

```diff
@@ -144,25 +144,32 @@ inline int na() {
   return NA_INTEGER;
 }
 
-// forward declaration
-
+// forward declarations
 typedef r_vector<double> doubles;
 
+template <>
+bool is_na(const double& x);
+
+template <>
+double r_vector<double>::operator[](const R_xlen_t pos) const;
+
 inline integers as_integers(sexp x) {
   if (TYPEOF(x) == INTSXP) {
     return as_cpp<integers>(x);
   } else if (TYPEOF(x) == REALSXP) {
     doubles xn = as_cpp<doubles>(x);
-    size_t len = (xn.size());
-    writable::integers ret = writable::integers(len);
-    for (size_t i = 0; i < len; ++i) {
+    R_xlen_t len = xn.size();
+    writable::integers ret(len);
+    for (R_xlen_t i = 0; i < len; ++i) {
       double el = xn[i];
-      if (!is_convertable_without_loss_to_integer(el)) {
+      if (is_na(el)) {
+        ret[i] = na<int>();
+      } else if (is_convertable_without_loss_to_integer(el)) {
```

Comment:
Is this a good opportunity to fix the spelling here (and where it's defined and the other place it's used)? To `is_convertible_without_loss_to_integer()`.